pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/api/ReportingService.kt

```diff
@@ -12,19 +12,35 @@ interface ReportingService {
 }
 
 data class ReportingRequest(
```

Comment:
I'd just have an empty `sealed class ReportingRequest` here, and let the events implement everything.

At most, it would require an `event: String`.